kokoro/gcp_ubuntu/releases/kokoro_build_release.sh

```diff
@@ -0,0 +1,129 @@
+export SCRIPT_WORKING_DIR="${pwd}"
+export SCRIPT_DIR=$(dirname "$0")
+export SCRIPT_PARENT_DIR="$(dirname -- "$SCRIPT_DIR")"
+export BUILD_SCRIPT=".$SCRIPT_WORKING_DIR/kokoro/gcp_ubuntu/kokoro_build.sh"
+source "${SCRIPT_DIR}/release_utils.sh"
+
+cd "${KOKORO_BUILD_RELEASE_DIR}"
+
+export LAST_GIT_TAG="$(git tag  \
+    | grep -E '^v[0-9]' \
+    | sort -V \
+    | tail -1 )"
+
+use_existing_tag=false
+
+# Do we already know what version we want to release?
+if [[ -z "${USE_TAG}" ]]; then
+  if [[ -z "${LAST_GIT_TAG}" ]]; then
+    err "No previous git tag found and it was not provided with USE_TAG env"
+  fi
+  VERSION="$(increment_tag_version ${LAST_GIT_TAG})"
+
+  log "Will create new version ${VERSION}"
+
+  if [ "$(git tag -l ${VERSION})" ]; then
+    err "ERROR! Tag for ${VERSION} already exists!"
+  fi
+
+  code=$(http_get_error_code "Accept: application/vnd.github.v3+json" \
+    "https://${GIT_RELEASES_USERNAME}:${GIT_PSW}@api.github.com/repos/google/dwh-assessment-extraction-tool/releases/tags/${VERSION}" )
+  if [ "$code" != "404" ]; then
+      err "ERROR! Release with ${VERSION} tag version name already exists or http failed! Http code is ${code}" 
+  fi
+else
+  VERSION="${USE_TAG}"
+  use_existing_tag=true
+
+  if [  -z "$(git tag -l ${VERSION})" ]; then
+    err "ERROR! Tag ${VERSION} is expected but does not exists exists!"
+  fi
+fi
+
+log "Version name ${VERSION} verified"
+
+# Run build and integration tests
+cd "${SCRIPT_WORKING_DIR}"
+log "Build script : ${BUILD_SCRIPT}, currend dir $(pwd)"
+bash "${BUILD_SCRIPT}"  
+
+# revert to initial state after running build script
+cd "${KOKORO_BUILD_RELEASE_DIR}"
+
+# create and register tag for this release if it was not manually provided
+if ! [[ use_existing_tag ]]; then
+  log "Create new tag"
+  git tag -a "${VERSION}" -m "${VERSION}"
+  git push "https://${GIT_RELEASES_USERNAME}:${GIT_PSW}@github.com/google/dwh-assessment-extraction-tool.git" "${VERSION}"
+fi
+
+log "Prepare release notes"
+
+output=$(http_post_check_status  "200" "Accept: application/vnd.github.v3+json" \
+  "https://${GIT_RELEASES_USERNAME}:${GIT_PSW}@api.github.com/repos/google/dwh-assessment-extraction-tool/releases/generate-notes" \
+  '{"tag_name":"'${VERSION}'","previous_tag_name":"'${LAST_GIT_TAG}'"}' )
+release_body="$(jq -r '.body'   <<< $output )"
+
+log "Create release"
+
+payload=$(
+  jq --null-input \
+     --arg tag "$VERSION" \
+     --arg name "$VERSION" \
+     --arg body "$release_body" \
+     '{ tag_name: $tag, name: $name, body: $body, draft: false }'
+)
+
+# Create new release as a draft
+response=$(http_post_check_status "201" "Accept: application/vnd.github.v3+json" \
+  "https://${GIT_RELEASES_USERNAME}:${GIT_PSW}@api.github.com/repos/google/dwh-assessment-extraction-tool/releases" "$payload" )
+
+log "Append zip file to the release"
+
+# Attach release binary file to the release
+release_id="$(jq -r '.id' <<< $response )"
```

Comment:
Some as above. If `${response}` contains spaces, this will cause problems. Hence, dynamic parameters should always be quoted:

```
release_id="$(jq -r '.id' <<< "${response}")"
```

Reply:
Thanks for your comment, truth must be told this code was not meant to be submitted for the review after first round, I needed to test this script on kokoro after adding additional functionality so it required me to submit the code.